drv/hisi_hpre.c

```diff
@@ -100,26 +100,31 @@ struct hisi_hpre_ctx {
 	struct wd_ctx_config_internal	config;
 };
 
-static bool is_hpre_bin_fmt(const char *data, int dsz, int bsz)
+static bool is_hpre_bin_fmt(char *dst, const char *src, int dsz, int bsz)
 {
-	const char *temp = data + dsz;
+	const char *temp = src + dsz;
 	int lens = bsz - dsz;
 	int i = 0;
 
+	if (!lens)
+		return true;
+
 	while (i < lens) {
-		if (temp[i] && !data[i])
-			return true;
+		if (temp[i] && !src[i])
+			break;
 		i++;
 	}
 
+	if (dst == src && i != lens)
```

Comment:
According to commit comments, the function of code snippet isn't be changed.
Where is the checking on dsz and bsz?
```
if (b_size == d_size || (dst == src && is_hpre_bin))
        return WD_SUCCESS;
```
So it seems the logic is changed. It's not consistent with the commit comments.